packages/smooth_app/lib/pages/history_page.dart

```diff
@@ -0,0 +1,45 @@
+import 'package:flutter/material.dart';
+import 'package:provider/provider.dart';
+import 'package:smooth_app/data_models/product_list.dart';
+import 'package:smooth_app/database/dao_product_list.dart';
+import 'package:smooth_app/database/local_database.dart';
+import 'package:smooth_app/pages/product/common/product_list_page.dart';
+
+class HistoryUpdatedSemaphore extends ChangeNotifier {
+  bool needsReload = true;
+  // Notify the history page that it needs to reload.
+  void notify() {
+    needsReload = true;
```

Comment:
I don't think you need this `bool`, the listeners should just sync up when they are notified

Reply:
I think that without it, the database will be hit on every `build` of the history page. I guess it might work, but it could also make the app really slow and resource intensive. WDYT?